lib/routes/gsau/news.ts

```diff
@@ -0,0 +1,158 @@
+import { DataItem, Route } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+import { load } from 'cheerio';
+import { generateRssItemForUnsupportedLink } from './utils/content';
+import { isSubdomainOfGsau } from './utils/domain';
+
+type NewsCategory = {
+    title: string;
+    description: string;
+};
+
+const NEWS_TYPES: Record<string, NewsCategory> = {
+    xxyw: {
+        title: '学校要闻',
+        description: '甘肃农业大学学校要闻',
+    },
+    xykx: {
+        title: '校园快讯',
+        description: '甘肃农业大学校园快讯',
+    },
+    tzgg: {
+        title: '通知公告',
+        description: '甘肃农业大学校内通知公告',
+    },
+    jzbg: {
+        title: '讲座报告',
+        description: '甘肃农业大学讲座报告信息',
+    },
+    jqgz: {
+        title: '近期关注',
+        description: '甘肃农业大学近期关注',
+    },
+    jyjx: {
+        title: '教育教学',
+        description: '甘肃农业大学教育教学新闻',
+    },
+    xsky: {
+        title: '学术科研',
+        description: '甘肃农业大学学术科研信息',
+    },
+    hzjl: {
+        title: '合作交流',
+        description: '甘肃农业大学合作交流信息',
+    },
+    mzgn: {
+        title: '每周甘农',
+        description: '甘肃农业大学周记总结',
+    },
+    mtnd: {
+        title: '媒体农大',
+        description: '相关对甘肃农业大学的媒体报道',
+    },
+};
+
+const handler: Route['handler'] = async (context) => {
+    const { category } = context.req.param();
+
+    if (!category || !NEWS_TYPES[category]) {
```

Comment:
Unnecessary check of `!category` since https://github.com/DIYgod/RSSHub/blob/3f536c8c102684fee4148e55e6c109de98b650b5/lib/routes/gsau/news.ts#L129 defines category as a compulsory path parameter, user who doesn't provide `category` will be rejected by middleware with https://github.com/DIYgod/RSSHub/blob/55b6afeed45e910e5fedfd0d5c0750df89572be7/lib/errors/types/not-found.ts#L1